Sources/Build/BuildDescription/ClangTargetBuildDescription.swift

```diff
@@ -238,23 +238,14 @@ public final class ClangTargetBuildDescription {
         }
 
         // Enable Clang module flags, if appropriate.
-        let enableModules: Bool
         let triple = self.buildParameters.triple
-        if toolsVersion < .v5_8 {
-            // For version < 5.8, we enable them except in these cases:
-            // 1. on Darwin when compiling for C++, because C++ modules are disabled on Apple-built Clang releases
-            // 2. on Windows when compiling for any language, because of issues with the Windows SDK
-            // 3. on Android when compiling for any language, because of issues with the Android SDK
-            enableModules = !(triple.isDarwin() && isCXX) && !triple.isWindows() && !triple.isAndroid()
-        } else {
-            // For version >= 5.8, we disable them when compiling for C++ regardless of platforms, see:
-            // https://github.com/llvm/llvm-project/issues/55980 for clang frontend crash when module
-            // enabled for C++ on c++17 standard and above.
-            enableModules = !isCXX && !triple.isWindows() && !triple.isAndroid()
-        }
-
+        // Swift is able to use modules on non-Darwin platforms because it injects its own module maps
+        // via vfs. However, nothing does that for C based compilation, and so non-Darwin platforms can't
+        // support clang modules.
+        // Modules are disabled for C++ though, see https://github.com/llvm/llvm-project/issues/55980
```

Comment:
I think the crash happens on non-Darwin and the justification comment should instead be

> on Darwin when compiling for C++, because C++ modules are disabled on Apple-built Clang releases

Reply:
If there's no crash on Darwin then we could pass `-fmodules` on Darwin, it just won't do anything for C++ sources because on Darwin `-fmodules` doesn't also enable `-fcxx-modules`.

Reply:
I think we should do that then, would simplify the logic here.